Menu Button: supporting overflow button

dkallus opened the pull request:
## Description
User can now show menu button as an overflow icon by setting a single attribute.

## Context
Section title and guidance module require a menu button with overflow icon. Showing overflow icon is already possible with existing implementation, however, user has to set `icon="overflow" priority="none" no-toggle-icon borderless` attributes and some additional styling. With this change, we can set a single attribute `variant="overflow"` to show the desired overflow icon.

I added 1 inline styling `style="border:0;"` to remove the border around the icon. Without this, we'll need a change to Skin.

Also, I added `expand-btn` class to the button since it's required for the menu to show/hide correctly.

I will change PR to 4.3.0 once branch is created.

## References
ebay-menu-button: support overflow icon #890

Depends on https://github.com/eBay/skin/pull/866 Menu Button: added generic class for toggling menu

## Screenshots
<img width="607" alt="overflow3" src="https://user-images.githubusercontent.com/844822/66081939-f6446300-e51d-11e9-8009-7e568dac7e20.png">

seangates commented on src/components/ebay-menu-button/template.marko:
```diff
@@ -15,35 +15,54 @@
+                "${baseClass}__button",
+                "expand-btn"
+            ]
+            style="border:0;"
```
I was initially concerned about this, but I see why you did this.

I guess the alternative was to add a special class `expand-btn--overflow` and create a new `style.less` file with that class and the `border: 0;` rule.

I'm not opposed to this given the desire to re-use as much as possible. This uses a lot less code and limits payload (albeit, it wouldn't have been a huge amount anyway).

seangates commented:
@dkallus So, developers have to remember to use `reverse` (if it's used on the right side)? Not a huge deal.

ianmcburnie commented:
`Also, I added expand-btn class to the button since it's required for the menu to show/hide correctly.`

I think we might want to change this situation. If not in this PR, then later in a refactor. What and where is the dependency?

dkallus commented:
@seangates Correc

> @dkallus So, developers have to remember to use `reverse` (if it's used on the right side)? Not a huge deal.

@seangates Correct. I'm fine with adding in reverse by default, but then there's no way for user to us not-reversed if they ever come across that scenario.

seangates commented:
@dkallus We could, in theory, just invert `reverse`, so that it works _opposite_ of their setting. 🤔

dkallus commented:
> `Also, I added expand-btn class to the button since it's required for the menu to show/hide correctly.`
> 
> I think we might want to change this situation. If not in this PR, then later in a refactor. What and where is the dependency?

@ianmcburnie 

> `Also, I added expand-btn class to the button since it's required for the menu to show/hide correctly.`
> 
> I think we might want to change this situation. If not in this PR, then later in a refactor. What and where is the dependency? I feel like using `icon-btn` is a better long term fix. Expand-btn is a very specific button, which made a lot of sense when menu was coupled to it's button, but not so much now.



> `Also, I added expand-btn class to the button since it's required for the menu to show/hide correctly.`
> 
> I think we might want to change this situation. If not in this PR, then later in a refactor. What and where is the dependency? I feel like using `icon-btn` is a better long term fix. Expand-btn is a very specific button, which made a lot of sense when menu was coupled to it's button, but not so much now.

@ianmcburnie It comes from `ebay-menu` styling. https://github.com/eBay/skin/blob/dcfcfc34f68d114aa1a045aa3143994e458ba3b6/src/less/menu-button/base/menu-button.less#L118-L121. We don't check if `.menu-button__button` is expanded, we check if `.expand-btn` is expanded.

ianmcburnie commented:
4.3.0 branch is now ready.

dkallus commented:
@agliga Can you take another look? I'm now using a single button and then switching between overflow vs default inside buttons' body. With this change, I also removed the browser overflow tests, since now we're using same button for existing flow vs overflow flow.

agliga commented:
> @agliga Can you take another look? I'm now using a single button and then switching between overflow vs default inside buttons' body. With this change, I also removed the browser overflow tests, since now we're using same button for existing flow vs overflow flow.

Much better!

ianmcburnie commented:
Let's make it reverse by default for overflow case.

dkallus commented:
@seangates I updated to use opposite value of "reverse" for overflow variant. Passing `reverse` attribute will correctly switch directions.

As for overflow example, I left overflow on the left side. I didn't find a simple way to align to right side without introducing a <div> wrapper.
